tests/testthat/test-age_helpers.R

```diff
@@ -0,0 +1,391 @@
+
+
+    # SQLite does not have a precise way to estimate age, so the age computation helper will throw a warning
+    # We here test that the warning is thrown, and filter out the known offending dates from the test
+    if (inherits(conn, "SQLiteConnection")) {
+
+      # Capture and check warning
+      test_ages <- tryCatch(
+        dplyr::mutate(test_ages, "age" = !!age_on_date("birth_date", "reference_date", conn)),
+        warning = function(w) {
+          expect_match(w$message, "Age computation on SQLite is not precise")
+          return(
+            suppressWarnings(dplyr::mutate(test_ages, "age" = !!age_on_date("birth_date", "reference_date", conn)))
+          )
+        }
+      )
+
+      # Remove offending dates
+      test_ages      <- dplyr::filter(test_ages,      .data$birth_date != !!as.numeric(as.Date("2000-07-28")))
+      reference_ages <- dplyr::filter(reference_ages, .data$birth_date != as.Date("2000-07-28"))
+
+    } else {
+
+      # Compute the age using the age function
+      test_ages <- dplyr::mutate(test_ages, "age" = !!age_on_date("birth_date", "reference_date", conn))
+
+    }
+
+    expect_equal(
+      dplyr::pull(test_ages,      "age"),
+      dplyr::pull(reference_ages, "age")
+    )
+
+    DBI::dbDisconnect(conn)
+  }
+})
+
+
+
+
+
+# Create a test data set that contains all birth dates in 2000, 1970, and 1940 (years with, or almost with, leap years)
+test_data <- c(
+  seq.Date(from = as.Date("2000-01-01"), to = as.Date("2000-12-31"), by = "1 day"),
+  seq.Date(from = as.Date("1970-01-01"), to = as.Date("1970-12-31"), by = "1 day"),
+  seq.Date(from = as.Date("1940-01-01"), to = as.Date("1940-12-31"), by = "1 day")
+) |>
+  tibble::tibble(birth_date = _) |>
+  dplyr::mutate(years_to_add = 1)
+
+
+test_that("add_years() works for positive input", {
+
+  for (conn in get_test_conns()) {
+
+    # Copy to the remote
+    test_ages <- dplyr::copy_to(conn, test_data, "test_age")
+    SCDB::defer_db_cleanup(test_ages)
+
+    # Compute the reference date using lubridate
+    reference_ages <- test_data |>
+      dplyr::mutate("first_birthday" = lubridate::`%m+%`(.data$birth_date, lubridate::years(1)))
+
+
+    # SQLite does not have a precise way to estimate time, so the time computation helper will throw a warning
+    # We here test that the warning is thrown, and filter out the known offending dates from the test
+    if (inherits(conn, "SQLiteConnection")) {
+
+      # Capture and check warning
+      test_ages <- tryCatch(
+        dplyr::mutate(test_ages, "first_birthday" = !!add_years("birth_date", 1, conn)),
+        warning = function(w) {
+          expect_match(w$message, "Time computation on SQLite is not precise")
+          return(suppressWarnings(dplyr::mutate(test_ages, "first_birthday" = !!add_years("birth_date", 1, conn))))
+        }
+      )
+
+
+      # Almost all dates in leap-years fail on SQLite
```

Comment:
This has to be addressed. It does provide a warning every time ... but is that sufficient?

Reply:
I don't think there is much else we can do.

We could suggest the user to use DuckDB as local alternative in the warning message?

We could also flat out throw an error if using SQLite

Reply:
Suggesting DuckDB could be a good compromise.